byway/src/main/java/com/google/sps/servlets/StopsServlet.java

```diff
@@ -59,7 +59,6 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     ArrayList<String> stops = gson.fromJson(request.getReader(), ARRAYLIST_STRING);
-    keyAsString = (String) request.getParameter("tripKey");
     trip = Trip.getTrip(datastore, keyAsString);
```

Comment:
Where is the value of `keyAsString` coming from? This should always come the request. You should go ahead and replace `trip` and `keyAsString` with local variables that are extracted based on the request.

Reply:
It was originally saved as a global variable and set during the get requests. I removed it because there were several internal server errors when trying to make a post request, which I'm assuming came from `request.getReader()` somehow. Since get requests usually come before post requests, I didn't think there was harm.

But then I remembered I had to do a similar things with my Servlet handling. the difference is that I got the key string before calling the reader, so I'm assuming the order had a difference. Now it contains local variables that store the Trip/tripKey as intended, while passing the trip Key for post requests too

Fixed